opennms-doc/guide-install/src/asciidoc/text/run-with-docker/introduction.adoc

```diff
@@ -0,0 +1,190 @@
+
+== Run with Docker
+
+Modern infrastructure allows you to deploy and run workloads in containers.
+With _{opennms-product-name}_ we provide and publish container images on link:https://hub.docker.com/u/opennms[DockerHub].
+
+NOTE: We don't install all available plugins in our published Docker image.
+      If you want ot customize and maintain your own image, you can find the _Dockerfiles_ in our link:https://github.com/OpenNMS/opennms/tree/develop/opennms-container[source repository].
+
+=== Objectives
+
+* Run _{opennms-product-name}_ using _Docker Compose_ with a basic setup and _PostgreSQL_ on your local system as a quickstart
+* Persist RRD files from _{opennms-product-name}_ and _PostgreSQL_ in a volume
+* Run and configure a _Minion_ in the stack and connect it to the _{opennms-product-name}_ instance using environment variables
+* Introduce a reference with all available configuration and mount conventions for more advanced setups
+
+=== Before you begin
+
+It is required you have at least the following components installed:
+
+* Current stable _Docker_ release installed, e.g. installed from link:https://docs.docker.com/[Docker Documentation]
+* Current stable _Docker Compose_ installed, e.g. installed from link:https://docs.docker.com/compose/install/[Docker Compose instructions]
+* You should have a basic knowledge about _Docker_, _Docker Compose_ with networking, persisting files and mounting directories
+
+=== Quickstart service stack
+
+// No section numbers for step-by-step guide
+:!sectnums:
+
+==== Step 1: Create service stack for PostgreSQL and _{opennms-product-name}_
+
+The first section describes how to setup _{opennms-product-name}_ service stack in a `docker-compose.yml` file.
+Create a project directory withg `mkdir opennms-horizon` and create inside a `docker-compose.yml` file with the following content:
+
+[source,yaml]
+----
+---
+version: '3'
+
+volumes:
+  data-postgres: {}<1>
+  data-opennms: {}<2>
+
+services:
+  database:<3>
+    image: postgres:12<4>
+    container_name: database<5>
+    environment:<6>
+      - TZ=Europe/Berlin
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+    volumes:<7>
+      - data-postgres:/var/lib/postgresql/data
+    healthcheck:<8>
+      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
+      interval: 10s
+      timeout: 30s
+      retries: 3
+
+  horizon:
+    image: opennms/horizon:25.0.0<9>
```

Comment:
I was not able to use maven properties version properties in source files, If don't want to maintain this, we need to add a placeholder and the user needs to replace it with the version he wants to install.